integration/simulation/devnetwork/sim_node.go

```diff
@@ -0,0 +1,199 @@
+package devnetwork
+
+import (
+	"fmt"
+	"math/big"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	gethlog "github.com/ethereum/go-ethereum/log"
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/common/log"
+	"github.com/obscuronet/go-obscuro/go/common/metrics"
+	"github.com/obscuronet/go-obscuro/go/config"
+	enclavecontainer "github.com/obscuronet/go-obscuro/go/enclave/container"
+	"github.com/obscuronet/go-obscuro/go/enclave/db/sql"
+	"github.com/obscuronet/go-obscuro/go/ethadapter"
+	hostcontainer "github.com/obscuronet/go-obscuro/go/host/container"
+	"github.com/obscuronet/go-obscuro/go/host/p2p"
+	"github.com/obscuronet/go-obscuro/go/host/rpc/clientrpc"
+	"github.com/obscuronet/go-obscuro/go/host/rpc/enclaverpc"
+	"github.com/obscuronet/go-obscuro/go/wallet"
+	"github.com/obscuronet/go-obscuro/integration"
+	"github.com/obscuronet/go-obscuro/integration/common/testlog"
+	"github.com/obscuronet/go-obscuro/integration/simulation/network"
+	"github.com/obscuronet/go-obscuro/integration/simulation/params"
+)
+
+// InMemNodeOperator represents an Obscuro node playing a role in a DevSimulation
+//
+// Anything a node operator could do, that we need to simulate belongs here, for example:
+// * start/stop/reset the host/enclave components of the node
+// * provide convenient access to RPC clients for the node
+// * it might even provide access to things like the database files for the node if needed
+//
+// Note: InMemNodeOperator will panic when things go wrong, we want to fail fast in sims and avoid verbose error handling in usage
+type InMemNodeOperator struct {
```

Comment:
would be useful to have the same concept for the l1 nodes as well. 
Stopping the l1 node, etc, is a useful thing to test as well